Correction of Various Documentation Issues

RKStrand opened the pull request:
Pull request overview
---------------------
 - Fixes #8277 
 - The purpose of this pull request is the implementation of a host of documentation fixes in various EnergyPlus documents.  Most of the changes were in the Input Output Reference, but there were changes in several other documents as well.  The issues dealt with are too numerous to repeat here.  Consult #8277 for a full checklist of what was resolved in this work.  A few typos were also corrected in the code (comment lines) and the IDD.  No changes were made to any logic.
 - Fixes #6974
 - Fixes #4306

Diffs expected for 1ZoneUncontrolledWithHysteresisPCM due to input file change.


### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

RKStrand committed:
Some of the changes for 8277

A good portion, but not all, of the changes requested as part of 8277 (or referred to 8277).  These are mostly documentation fixes.  The few code or IDD fixes are simply correcting typos.

RKStrand committed:
Merge branch 'develop' into 8277DocumentationIssues2021a

RKStrand committed:
Update to Interior Window Output

Update to variables that can be output for interior windows and other minor edits.

RKStrand committed:
Remaining changes to IORef

Last changes required to IO Reference to address issues related to extensible IDD and many other edits.

mjwitte commented on idd/Energy+.idd.in:
```diff
@@ -5,6 +5,14 @@
 ! Lines in EnergyPlus input files (and IDD) are limited to 500 characters.
 !
+!
+! Note on IDD Processing/Usage
```
This entire block of comments also appear in the I/O Reference. I would just copy from the IDD to update the entire block (which may be out of date in other ways0.

RKStrand replied:
Fixed--see next commit.

mjwitte commented on doc/input-output-reference/src/overview/group-advanced-surface-concepts.tex:
```diff
@@ -2294,7 +2292,7 @@ \subsubsection{Inputs}\label{inputs-13}
 \paragraph{Field(s): Surface \textless{}1 thru x\textgreater{} Name}\label{fields-surface-1-thru-x-name}
 
-The remaining fields are used to name the \textbf{\hyperref[buildingsurfacedetailed]{BuildingSurface:Detailed}} objects that are associated with the exterior naturally vented cavity. These are the underlying heat transfer surfaces and are defined elsewhere in the input file. These surfaces should all specify OtherSideConditionsModel as their exterior environment. The input object can currently ac ...
+The remaining fields are used to name the \textbf{\hyperref[buildingsurfacedetailed]{BuildingSurface:Detailed}} objects that are associated with the exterior naturally vented cavity. These are the underlying heat transfer surfaces and are defined elsewhere in the input file. These surfaces should all specify OtherSideConditionsModel as their exterior environment. The input object can currently ac ...
```
This object is extensible, not limited to 10 surfaces. Also, the introductory paragraph for this object, SurfaceProperty:ExteriorNaturalVentedCavity, needs editing, it still says "(although if you need to use more than 10 surfaces, then the IDD will need to be extended)."

RKStrand replied:
Wow--good catch.  I thought I got all of those but clearly missed this one.  Will be fixed in the next commit.

mjwitte commented on doc/input-output-reference/src/overview/group-electric-load-center-generator.tex:
```diff
@@ -524,7 +524,7 @@ \subsubsection{Outputs}\label{outputs-1-008}
 \subsection{ElectricLoadCenter:Generators}\label{electricloadcentergenerators}
 
-The ElectricLoadCenter:Generators object is used to provide a list of the generators to include in the simulation. The list includes the names and types of all the generators along with separate availability schedules, the rated power output, and thermal-to-electrical power ratio for each. Sets of five input fields are repeated for each generator. If more than 30 generators are needed, EnergyPlus ...
+The ElectricLoadCenter:Generators object is used to provide a list of the generators to include in the simulation. The list includes the names and types of all the generators along with separate availability schedules, the rated power output, and thermal-to-electrical power ratio for each. Sets of five input fields are repeated for each generator. If more than 30 generators are needed, EnergyPlus ...
```
"If more than 30 generators are needed, EnergyPlus will auto-extend to suit the needs but other interfaces (such as the IDF Editor) may not. The user can always modify the Energy+.idd file to accommodate the extra fields necessary, but it is likely the next release of EnergyPlus will overwrite any user modifications." I would replace this with the standard This object is extensible sentence.

RKStrand replied:
Good call--will be fixed in the next commit.

mjwitte commented on doc/input-output-reference/src/overview/group-refrigeration.tex:
```diff
@@ -1055,7 +1055,7 @@ \subsubsection{Field: Insulated Floor U-Value}\label{field-insulated-floor-u-val
 The floor themal transmittance (in W/m\(^{2}\)-K). This value has a default value of 0.3154. (This corresponds to R18 in Archaic American Insulation Units.~ To convert other R-values to thermal transmittance, divide 5.678 by the R-value.~ For example, R15 is 0.3785 W/m\(^{2}\)-K and R5 is 1.136 W/m\(^{2}\)-K.)
 
-\textbf{\emph{THE REMAINING 12 FIELDS FOR THE WALK-IN COOLER MUST BE REPEATED FOR EACH ZONE WHICH IS IN CONTACT WITH A WALK-IN WALL, CEILING, OR DOOR. The IDD includes fields for 3 zones, but can be extended by repeating the last 12 values in the object.}}
+\textbf{\emph{THE REMAINING 12 FIELDS FOR THE WALK-IN COOLER MUST BE REPEATED FOR EACH ZONE WHICH IS IN CONTACT WITH A WALK-IN WALL, CEILING, OR DOOR.  This field is extensible, so additional fields of this type can be added to the end of this object by repeating the last 12 values in the object.}}
```
This *object* is extensible.
Actually, most of the places that say "This field is extensible..." probably should say "This object is extensible" I guess saying "this field" is ok for places where it's just a single extensible field.

RKStrand replied:
Generally my pattern was if it was a single field I used the term field.  If it was a group of fields that were repeated, I adjusted the language to say that.  Are you ok with that?  If not, I'll go through and change them.

RKStrand replied:
I'm in the process of changing everything to "This object is extensible..." and cleaning up how these some of these show up in the docs.  So ignore the question from the above comment.

mjwitte commented on doc/input-output-reference/src/overview/group-demand-limiting-controls.tex:
```diff
@@ -70,15 +70,15 @@ \subsubsection{Inputs}
 The name of a DemandManager object defined elsewhere in the input file.
 
-DemandManagers are listed by pairs of data items: \emph{Demand Manager Type} and \emph{Demand Manager Name}. Ten managers are accommodated  in the list by default. The IDD specification, however, is extensible and additional pairs may be added by directly editing the IDD.
+DemandManagers are listed by pairs of data items: \emph{Demand Manager Type} and \emph{Demand Manager Name}. These two fields are extensible, so up to 10 additional pairs of these fields can be added to the end of this object.
```
This object is extensible, so ~up to 10~ additional pairs of these fields ....

RKStrand replied:
Got it--fixed in the next commit.

mjwitte commented on doc/input-output-reference/src/overview/group-solar-collectors.tex:
```diff
@@ -666,7 +666,7 @@ \subsection{SolarCollector:UnglazedTranspired}\label{solarcollectorunglazedtrans
 This object is used to model unglazed transpired solar collectors (UTSC) used to condition outdoor air. These collectors are generally used to heat air drawn through perforated absorbers that are heated by the sun and also recover heat conducted out through the underlying wall. The SolarCollector:UnglazedTranspired object represents a single collector attached to one or more \hyperref[buildingsur ...
 
-The area and orientation of the collector is obtained from \hyperref[buildingsurfacedetailed]{BuildingSurface:Detailed} objects, which are referenced by name. Although the collector surface itself is slightly detached from the underlying building wall (or roof), no additional surface object is needed to represent the collector itself. When modeling transpired collectors, it is important to consid ...
+The area and orientation of the collector is obtained from \hyperref[buildingsurfacedetailed]{BuildingSurface:Detailed} objects, which are referenced by name. Although the collector surface itself is slightly detached from the underlying building wall (or roof), no additional surface object is needed to represent the collector itself. When modeling transpired collectors, it is important to consid ...
```
For this object, the Surface <#> Name field still says "can currently accommodate up to ten surfaces, but it is extensible." I would change that to your standard extensible sentence.

RKStrand replied:
Fixed in next commit

mjwitte reviewed (commented):
@RKStrand Some initial comments. If you scroll through the changes page on the PR, you'll see some warnings in yellow and at the end there are some errors reported (which you did not introduce, I think I actually did those). If you can clean those up here, that would be good.

mjwitte commented on doc/input-output-reference/src/overview/group-pumps.tex:
```diff
@@ -100,11 +100,13 @@ \subsubsection{Inputs}\label{inputs-037}
 
-For VFD Control Type = PressureSetPointControl, this field defines the minimum allowable RPM, or the lower bound of pump speed to use when determining the required pump speed. For VFD Control Type = ManualControl, this field is not interpreted. The value of the schedule is RPM.
+For VFD Control Type = ManualControl, this field defines the minimum allowable RPM, or
+the lower bound of pump speed to use when determining the required pump speed. For VFD Control Type = PressureSetPointControl, this field is not interpreted. The value of the schedule is RPM.
 
 \paragraph{Field: Maximum RPM Schedule}\label{field-maximum-rpm-schedule}
 
-For VFD Control Type = PressureSetPointControl, this field defines the maximum allowable RPM, or the upper bound of pump speed to use when determining the required pump speed. For VFD Control Type = ManualControl, this field is not interpreted. The value of the schedule is RPM.
+For VFD Control Type = ManualControl, this field defines the maximum allowable RPM, or
+the upper bound of pump speed to use when determining the required pump speed. For VFD Control Type = PressureSetPointControl, this field is not interpreted. The value of the schedule is RPM.
```
Revert these. It was correct before. See [comment here](https://github.com/NREL/EnergyPlus/issues/7006#issuecomment-863670094).

RKStrand replied:
Reverted in next commit.

mjwitte commented on doc/ems-application-guide/src/user-defined-component-models/plant-component.tex:
```diff
@@ -119,7 +123,7 @@ \subsection{Ambient Zone}\label{ambient-zone-001}
   An actuator called ``Component Zone Internal Gain'' with the control type ``Sensible Heat Gain Rate,'' in {[}W{]}, is available.~ This can be used for purely convective sensible heat gains (or losses) to a zone.
 \item
-  An actuator called ``Component Zone Internal Gain'' with the control type ``Return Air Heat Gain Rate,'' in {[}W{]}, is available.~ This can be used for purely convective sensible heat gains (or losses) to the return air duct for a zone.
+  An actuator called ``Component Zone Internal Gain'' with the control type ``Return Air Heat Sensible Gain Rate,'' in {[}W{]}, is available.~ This can be used for purely convective sensible heat gains (or losses) to the return air duct for a zone.
```
Thanks for the thorough review and updates on the EMS actuators!

RKStrand replied:
Thanks--this one would have been easy to overlook!

mjwitte reviewed (commented):
@RKStrand Lots of good stuff here. A few more comments to complete the initial review.

mitchute commented on doc/input-output-reference/src/overview/group-surface-construction-elements.tex:
```diff
@@ -720,6 +718,58 @@ \subsubsection{Inputs}\label{materialpropertyphasechangehysteresis-inputs}
+
+The \hyperref[materialpropertyphasechangehysteresis]{MaterialProperty:PhaseChangeHysteresis} object also includes the following outputs.  The Conduction Finite Difference solution algorithm uses a finite difference solution technique where the surfaces are divided into a nodal arrangement. These outputs are specific to Conduction Finite Difference solution.
+
+The following output variables are applicable to all opaque heat transfer surfaces when using Solution Algorithms ConductionFiniteDifference.  Note that the "X" in the list and descriptions below must be replaced by a number that indicates the node at which the variables are being reported.  So, for example, to report the surface temperature for node 7, one would use "CondFD Surface Temperature N ...
```
If you're already making changes, change the ```"``` to "``" (two backticks for the left side) and "''" (two apostrophes for the right side).

RKStrand replied:
Got it--actually ``them'' because there were two copies of this in this file.  Should be fixed in the next commit.

mitchute commented on doc/input-output-reference/src/overview/group-zone-forced-air-units.tex:
```diff
@@ -2938,9 +2938,9 @@ \subsubsection{Inputs}\label{inputs-10-024}
 
 The name of the third air chiller that will be used to meet the zone cooling load.
```
It doesn't look like you caused this, but can you address this warning? Are there plans elsewhere to fix the other LaTeX build issues that have surface recently?

Myoldmopar replied:
If this is in reference to the build warnings that pop up at the bottom of the Files Changed tab on all the PRs, those are now addressed in develop.

mitchute reviewed (commented):
Lot's of good stuff here. There's too much in the original issue to go line by line checking this, but I'm OK with the changes. I think there may be a few other comments left so we can get this in once that's all wrapped up.

Myoldmopar commented:
With the PR I just merged about the IORef (#8836) in develop now, this branch should have develop pulled up before merging.  @RKStrand do you have any additional work to make on this?  If not I'll merge develop and push it up.

@mjwitte @mitchute do you have any additional reviewing to do on this or is it ready if CI comes back clean?

RKStrand commented:
@Myoldmopar I'm still working through the review comments.  Might be done and ready for the next review later today, but I'm not really sure.

RKStrand committed:
Resolution of Review Comments

Additional changes were made based on review comments received from other development team members.  Should be ready for the next review now.

RKStrand committed:
Merge branch 'develop' into 8277DocumentationIssues2021a

RKStrand commented:
@Myoldmopar @mjwitte @mitchute Ok, I think I have addressed all of the comments above and brought develop into this branch as well.  Hopefully everything comes back clean now.  Let me know if there is anything else that needs to be done here.  Thanks!

mjwitte reviewed (approved):
@RKStrand Changes to address comments look good.
